src/app/SearchData.js

```diff
@@ -66,17 +67,19 @@ const SearchData: DataTableEntry<SearchProps> = {
 
     const searchQuery = query.q;
     let trimmedSearchQuery;
-    let searchResults = Promise.resolve({ features: [] });
+    let searchResults: Promise<SearchResultCollection> | SearchResultCollection = {
+      features: [],
+    };
 
     if (searchQuery && (trimmedSearchQuery = searchQuery.trim())) {
```

Comment:
0: Why not move trimming into if? I can't see where it is used.

Reply:
The trimmed search query is used when creating the request in line 75. It simply checks if the string is not empty for which we do not need to make a request to Photon.

Reply:
I don't the reason for a local variable with `let` then.

Reply:
I meant 'used elsewhere' above, sorry.

Reply:
This would make the code more complex IMO due to more nesting.

``` js
if (searchQuery) {
  let trimmedSearchQuery = searchQuery.trim()

  if (trimmedSearchQuery) {
    searchResults = (isServer ? searchPlaces : searchPlacesDebounced)(trimmedSearchQuery, {
      lat: parseFloat(query.lat),
      lon: parseFloat(query.lon),
    });

    // Fetch search results when on server. Otherwise pass (nested) promises as props into app.
    searchResults = isServer ? await searchResults : searchResults;
  }
}
```

Compared to this:

``` js
let trimmedSearchQuery;

if (searchQuery && (trimmedSearchQuery = searchQuery.trim())) {
  searchResults = (isServer ? searchPlaces : searchPlacesDebounced)(trimmedSearchQuery, {
    lat: parseFloat(query.lat),
    lon: parseFloat(query.lon),
  });

  // Fetch search results when on server. Otherwise pass (nested) promises as props into app.
  searchResults = isServer ? await searchResults : searchResults;
}
```

Reply:
Why no early returns?

Reply:
Good point! PR is already merged. But I will maybe add it to another PR (… living on the edge and so forth).